Whitelist-Dapp/hardhat-tutorial/contracts/Whitelist.sol

```diff
@@ -0,0 +1,39 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+
+contract Whitelist {
+
+    // Max number of whitelisted addresses allowed
+    uint8 public maxWhitelistedAddresses;
+
+    // Create a mapping of whitelistedAddresses
+    // if an address is whitelisted, we would set it to true, it is false by default for all other addresses.
+    mapping(address => bool) public whitelistedAddresses;
+
+    // numAddressesWhitelisted would be used to keep track of how many addresses have been whitelisted
+    // NOTE: Don't change this variable name, as it will be part of verification
+    uint8 public numAddressesWhitelisted;
+
+    // Setting the Max number of whitelisted addresses
+    // User will put the value at the time of deployment
+    constructor(uint8 _maxWhitelistedAddresses) {
+        maxWhitelistedAddresses =  _maxWhitelistedAddresses;
+    }
+
+    /**
+        addAddressToWhitelist - This function adds the address of the sender to the
+        whitelist
+     */
+    function addAddressToWhitelist() public {
```

Comment:
Add onlyOwner modifier to block external and illegitimate attempt

Reply:
Or admin list if there is multiple addresses working for this NFT project

Reply:
In this project this is a public whitelist then I am wondering if is it useful to add an owner. 
We can just add require(....) with the owner if in the future we want to create a function to delete member in the whitelist.